engine/src/main/java/org/hibernate/search/backend/NumRangeQuery.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend;
+
+/**
+ * @author Martin Braun
+ */
+public class NumRangeQuery implements DeletionQuery {
```

Comment:
This is because of the choice to only support special Queries and not all Queries. And for those 2 or 3 that we will need to delete by we can easily create our own version that's much much easier to serialize.